### PR TITLE
fix facet channel sort with transform

### PR DESCRIPTION
**Pull request by mbostock**
Fixes #840.

### CODE REVIEW

**Review by Fil — APPROVED** [review]
I've checked that it works well in conjunction with #1315.
Build: https://observablehq.com/@observablehq/plot-sort-1315-1316

File: src/mark.js [hunk 1]
```diff
@@ -78,9 +78,11 @@ export class Mark {
   initialize(facets, facetChannels) {
     let data = arrayify(this.data);
     if (facets === undefined && data != null) facets = [range(data)];
+    const originalFacets = facets;
     if (this.transform != null) ({facets, data} = this.transform(data, facets)), (data = arrayify(data));
+    if (facets !== undefined) facets.original = originalFacets; // needed up read facetChannels
```
Comment (mbostock) [inline src/mark.js:83]: “needed up read facetChannels” 🤔

Comment (Fil) [reply src/mark.js:83]: "needed to reindex the facet channel" (?)
or remove

Comment (mbostock) [reply src/mark.js:83]: Ah ha. I think I meant “needed **to** read facetChannels”.

**Pull request merged**